doc/languages-frameworks/haskell.section.md

```diff
@@ -1,7 +1,397 @@
-# Haskell {#haskell}
+# Haskell {#sec-haskell}
 
-The documentation for the Haskell infrastructure is published at
-<https://haskell4nix.readthedocs.io/>. The source code for that
-site lives in the `doc/` sub-directory of the
-[`cabal2nix` Git repository](https://github.com/NixOS/cabal2nix)
-and changes can be submitted there.
+The Haskell infrastructure in nixpkgs has two main purposes: The
+primary purpose is to provide a Haskell compiler and build tools
+as well as infrastructure for packaging Haskell-based packages.
+
+The secondary purpose is to provide support for Haskell development
+environment including prebuilt Haskell libraries. However, in this
+area sacrifices have been made due to self-imposed restrictions in
+nixpkgs, to lessen the maintenance effort and improve performance.
+Therefore it may be advantageous to use an alternative to the
+Haskell infrastructure in nixpkgs for development environments in
+many cases. The main limitations are that we only provide first-class
+support for the default compiler (currently GHC 8.10.4) and usually
+only provide a default and (if different) the latest version of
```

Comment:
latest being hackage (!= hackage) ?

Reply:
it's explained later on so I understand that if the package is on stackage, then the version is not necessarily the latest ?

Reply:
Yeah, I want to explain this in more detail in the section “Package Set Generation”. Basically we use the latest version of every package from hackage by default unless it is part of stackage or there has been a pin manually set up. In those cases we use the manual version constraint or the one from stackage to get the version of the standard attribute and additionally generate an attribute `package-name_A_B_C_D` which points to the latest version from hackage.